src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -46,21 +49,54 @@ DebugInitialize()
     const size_t environmentVariableValueSize = 2;
     std::wstring environmentVariableValue(environmentVariableValueSize, '\0');
 
-    if (GetEnvironmentVariable(L"ASPNETCORE_MODULE_DEBUG", environmentVariableValue.data(), environmentVariableValueSize) == environmentVariableValueSize - 1)
+    try
     {
-        try
-        {
-            const auto value = std::stoi(environmentVariableValue);
+        const auto value = std::stoi(Environment::ExpandEnvironmentVariables(L"%ASPNETCORE_MODULE_DEBUG%"));
 
-            if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
-            if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
-            if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
-            if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
-        }
-        catch (...)
+        if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
+        if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
+        if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
+        if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+    }
+    catch (...)
+    {
+        // ignore
+    }
+
+    try
+    {
+        const auto debugOutputFile = Environment::ExpandEnvironmentVariables(L"%ASPNETCORE_MODULE_DEBUG_FILE%");
+
+        if (!debugOutputFile.empty())
         {
-            // ignore
+            g_logFile = CreateFileW(debugOutputFile.c_str(),
+                FILE_GENERIC_WRITE,
+                FILE_SHARE_READ | FILE_SHARE_WRITE,
+                nullptr,
+                OPEN_ALWAYS,
+                FILE_ATTRIBUTE_NORMAL,
+                nullptr
+            );
+
+            if (g_logFile != INVALID_HANDLE_VALUE)
+            {
+                DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_FILE;
+            }
         }
+
+    }
+    catch (...)
+    {
+        // ignore
+    }
+}
+
+VOID
+DebugStop()
+{
+    if (IsEnabled(ASPNETCORE_DEBUG_FLAG_FILE))
```

Comment:
check whether the handle is InvalidHAndle instead of the flag, as the CreateFile may fail.